ifacemonitor/iface_monitor.go

```diff
@@ -136,8 +137,8 @@ readLoop:
 }
 
 func (m *InterfaceMonitor) isExcludedInterface(ifName string) bool {
-	for _, name := range m.InterfaceExcludes {
-		if ifName == name {
+	for _, nameExp := range m.InterfaceExcludes {
+		if nameExp.Match([]byte(ifName)) {
```

Comment:
I think there's a change of behaviour here for existing config.  With config `xyz,abc`, previously interface names had to be exactly `xyz` or `abc`.  But now, if I understand the regexes correctly, it will match any interface names that _include_ `xyz` or `abc`.

Am I right?  If so, is that a problem?

Reply:
> Am I right? If so, is that a problem?

Yes, you are correct @neiljerram! Very good point. 

This would be a problem because it would break backwards compatibility, i.e. with any Calico install where someone has `FELIX_INTERFACEEXCLUDE` set to some values where they expect only exact matches. For example, in an existing Calico install with something like `FELIX_INTERFACEEXCLUDE=veth1` I would want it only exclude an exact match of an interface with name `veth1` and not something with a name `veth11`. 

One solution to this obstacle is have a clear separation between values in the _old format_ (non-regexp exact match values) and values in the _new format_ (regexp values). 

Had a quick chat with @caseydavenport and a simple format could be: 
- Old values remain in the format they are now
- New values must be delimited by `/` on both ends to symbolize it is to be treated as a regex 

Example: 
- The below will exclude an exact interface name of `veth0`
```
FELIX_INTERFACEEXCLUDE=veth0
```
- The below will exclude an exact interface name of `veth0` 
```
FELIX_INTERFACEEXCLUDE=/^veth0$/
```
- The below will exclude (1) any interface name that begins with `veth` (e.g. `veth0`, `veth1`, ... etc) and any interface name with `kube-ipvs` as a substring (e.g. `prefix-kube-ipvs`, `kube-ipvs-suffix`, ``prefix-kube-ipvs-suffix`)
```
FELIX_INTERFACEEXCLUDE=/^veth/,/kube-ipvs/
```


Thoughts?

Reply:
Yes, that works and sounds OK to me.  Another way would be 
```
		regexpValues = append(regexpValues, regexp.MustCompile("^" + strValue + "$"))
```
in other words require the regex always to match the whole name.

But probably the explicit `/.../` syntax is better because it also avoids another possible back-compatibility problem, if it's possible for any existing settings to include regex metacharacters.

Reply:
> But probably the explicit /.../ syntax is better because it also avoids another possible back-compatibility problem, if it's possible for any existing settings to include regex metacharacters.

The existing validation for the interface exclude parameter only allows `[a-zA-Z0-9_-]`. Looking at the regexp constructs it does not look like `-` or `_` symbol would get us in trouble because they would just be interpreted as literals. 

Still, using the `/` on both ends felt like a nice clean way to communicate to users what to expect when looking at the values.